### DISCUSSION TITLE
Updated path for the Asset manifest file?

### DISCUSSION THREAD

**Discussion by DavidGarciaCat** [sonata-project/SonataAdminBundle#8165] [Q&A] [answered]
Good day,

Working on a new project, the customer's hosting requires the "public" folder to be renamed as "web".

Based on the Symfony website, the procedure is to set it as part of the `composer.json` file:

https://symfony.com/doc/current/configuration/override_dir_structure.html#override-the-public-directory

```json
{
    "...": "...",
    "extra": {
        "...": "...",
        "public-dir": "web",
    }
}
```

After installing the `Sonata Admin Bundle`, I get this error message:

> An exception has been thrown during the rendering of a template ("Asset manifest file ".../public/build/manifest.json" does not exist. Did you forget to build the assets with npm or yarn?").

Is there a way to change the Bundle's configuration to use the "web" folder instead of the "public" folder when running the `composer install -o` command, so the asset resources are installed into the expected folder?

Thanks,

**Comment by zyberspace** [comment]
Hi,

this is very likely not an issues with sonata admin but with the installed webpack encore bundle.

You should have a configuration file under `config/packages/webpack_encore.yaml`. There you need to update the `output_path` and the `json_manifest_path` (the later is causing your error).

https://github.com/symfony/recipes/blob/main/symfony/webpack-encore-bundle/2.0/config/packages/webpack_encore.yaml

**Reply by DavidGarciaCat** [reply]
👍 I'll check it in a moment and post an update here - thanks for the tip!

**Reply by DavidGarciaCat** [reply, marked as answer]
Three updates needed:

```diff
// webpack.config.js

Encore
    // directory where compiled assets will be stored
-    .setOutputPath('public/build/')
+    .setOutputPath('web/build/')
```

```diff
# config/packages/webpack_encore.yaml

webpack_encore:
    # The path where Encore is building the assets - i.e. Encore.setOutputPath()
-    output_path: '%kernel.project_dir%/public/build'
+    output_path: '%kernel.project_dir%/web/build'
```

```diff
# config/packages/webpack_encore.yaml

framework:
    assets:
-        json_manifest_path: '%kernel.project_dir%/public/build/manifest.json'
+        json_manifest_path: '%kernel.project_dir%/web/build/manifest.json'
```
